--- a/testsuite/python_support/utils.py
+++ b/testsuite/python_support/utils.py
@@ -434,8 +434,7 @@
             if any(f.endswith(ext) for ext in [".adb", ".ads"]):
                 langs.add("Ada")
 
-        # Generate a project file to build the given mains. Do a static build
-        # (the default) to improve the debugging experience.
+        # Generate a project file to build the given mains.
         with open("gen.gpr", "w") as f:
 
             def fmt_str_list(strings: List[str]) -> str:
@@ -446,7 +445,9 @@
                 source_dirs=fmt_str_list(source_dirs),
                 main_sources=fmt_str_list(gpr_mains),
             ))
-        run("gprbuild", "-Pgen", "-q", "-p")
+        run("gprbuild", "-Pgen", "-q", "-p",
+            '-XLIBRARY_TYPE=relocatable',
+            '-XXMLADA_BUILD=relocatable')
 
         # Now run all mains. If there are more than one main to run, print a
         # heading before each one.
--- a/testsuite/tests/misc/link_two_libs/test.py
+++ b/testsuite/tests/misc/link_two_libs/test.py
@@ -35,15 +35,12 @@
         manage_py,
         'make',
         '-vnone',
-
-        # To simplify this test (in particular environment setup), do a static
-        # link.
-        '--library-types=static',
+        '--library-types=relocatable',
     )
 
 # Build a program that uses both and run it
 subprocess.check_call(['gprbuild', '-q', '-Pmain.gpr', '-p',
-                       '-XLIBRARY_TYPE=static'])
+                       '-XLIBRARY_TYPE=relocatable'])
 subprocess.check_call([os.path.join('obj', 'main')])
 
 print('Done')
--- a/testsuite/testsuite.py
+++ b/testsuite/testsuite.py
@@ -191,7 +191,7 @@
                 gargs.append('--subdirs=gnatcov')
                 cargs.extend(['-fdump-scos', '-fpreserve-control-flow'])
 
-            for build in ('static', 'relocatable'):
+            for build in ('relocatable'):
                 run(
                     f'Langkit support - build {build}',
                     ['gprbuild'] + gargs +
